eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/FlatCollectIterable.java

```diff
@@ -78,6 +79,24 @@ public V detect(Predicate<? super V> predicate)
     }
 
     @Override
+    public Optional<V> detectOptional(Predicate<? super V> predicate)
+    {
+        V[] result = (V[]) new Object[1];
+        Iterate.detectOptional(this.adapted, each -> {
```

Comment:
The call to detectOptional is on the elements of type T before they are
flattened to V. If the iterable itself is null, we'll get a NPE simply by
iterating over null. Is there something else accomplished by the call?
On Mon, Aug 22, 2016 at 10:08 AM Hiroshi Ito <notifications@github.com>
wrote:

> In
> eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/FlatCollectIterable.java
> <https://github.com/eclipse/eclipse-collections/pull/136#discussion_r75682240>
> :
>
> > @@ -78,6 +79,24 @@ public V detect(Predicate<? super V> predicate)
> >      }
> >
> >      @Override
> > +    public Optional<V> detectOptional(Predicate<? super V> predicate)
> > +    {
> > +        V[] result = (V[]) new Object[1];
> > +        Iterate.detectOptional(this.adapted, each -> {
>
> This call is actually logically significant in a way that it should throw
> NullPointerException when it detects the value which is null. Throwing
> NullPointerException is handled within Iterate.detectOptional(), when
> Optional.of() is called.
>
> Having said that, we could change the detect() side as you suggest, but
> maybe keeping as is makes the structure consistent. Let me know.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/eclipse-collections/pull/136/files/0717baa7b61180c45ada084621d2cde845093825..7e18d3899e46ae223985eda4e753ed2da9c6dea5#r75682240>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAO6Ir3wcpt49YDcX0CR9yaHuhkxHykZks5qiatygaJpZM4Jol8s>
> .
>